### PR TITLE
add style guide

### PR DESCRIPTION
**Pull request by smart-alek**
Captures specific style preferences for our documentation.

### CODE REVIEW

**Comment by michelle-luna** [comment]
I feel like this belongs in the section for style that we have in the readme, rather than another document, what do you think?

**Comment by smart-alek** [comment]
For now, I think that makes sense. But once this guide starts getting longer, it makes more sense to me to create a separate file for style preferences. That limits the readme content to technical instructions on how to set up the repo and create documents.

Does that argument convince you? And if so, what do you think the threshold should be for pulling it into its own file?

**Comment by smart-alek** [comment]
@michelle-luna I sort of went to town on some of these. Take a look and veto what you'd like. Explanation of the recent changes is in the commit messages but ping me if you have questions about anything.

File: README-local-development.md [hunk 1]
```diff
@@ -38,6 +34,70 @@ Jekyll will build the site and start a web server, which can be viewed in your b
...
+title: "Your Doc Title"
+short-title: "Short Title"
+categories: [category-slug]
+order: 10
+---
+```
+
+`layout` describes visual settings shared across our docs. `title` will appear at the top of your article and appear in hyphenated form for the URL. `short-title` will display in the sidebar on the left underneath the article's category.
+
+The list of available categories can be found in [categories.yml](https://github.com/circleci/circleci-docs/blob/master/jekyll/_data/categories.yml).
```
Comment (michelle-luna) [inline README-local-development.md]: hmm, we aren't using this categories.yml convention in 2.0 docs anymore 
anymore. So, we should say that and explain that navigation links are manually added to 
the landing pages and that this is a decision the writer must make. (Or that the docs leads can make for the contributor).

File: README-local-development.md [hunk 2]
```diff
@@ -38,6 +34,70 @@ Jekyll will build the site and start a web server, which can be viewed in your b
...
+
+`order` specifies the article's placement within its category. You may need to change the `order` of other articles to get your article to appear where you want within a category. Best practice is to use multiples of 10; this makes it easy to put a new doc between two others.
+
+### Headings & Tables of Contents
+
+Jekyll will automatically convert your article's title into a level one heading (#), so we recommend using level two (##), level three (###) and level four (####) headings when structuring your article.
+
+If your article gets too long, you can add a table of contents with the reference name "toc":
```
Comment (michelle-luna) [inline README-local-development.md]: I would say 'if your article has more than three headings after the H1, then please use a TOC.

File: README-local-development.md [hunk 3]
```diff
@@ -38,6 +34,70 @@ Jekyll will build the site and start a web server, which can be viewed in your b
...
+
+```
+# Not in the TOC
+{:.no_toc}
+```
+
+# Submitting Pull Requests
 
-Now that you have a working local environment, please follow our [Contributing Guide](CONTRIBUTING.md) to make and submit changes.
+Once you are finished with your changes, please follow our [Contributing Guide](CONTRIBUTING.md) to submit a pull request.
```
Comment (michelle-luna) [inline README-local-development.md]: nit: Once -> After

File: README-local-development.md (hunk 1 above)
Comment (smart-alek) [reply README-local-development.md]: Does that mean we're also not using the `order` convention?

Comment (smart-alek) [reply README-local-development.md]: Good catch, though! :)

File: README-local-development.md (hunk 2 above)
Comment (smart-alek) [reply README-local-development.md]: Nice; less emotional impact. What even is "too long", anyways.

**Pull request merged**